parser/src/main/kotlin/com/linkedin/dex/parser/AnnotationUtils.kt

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) LinkedIn Corporation. All rights reserved. Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.dex.parser
+
+import com.linkedin.dex.spec.AnnotationItem
+import com.linkedin.dex.spec.AnnotationSetItem
+import com.linkedin.dex.spec.AnnotationsDirectoryItem
+import com.linkedin.dex.spec.ClassDefItem
+import com.linkedin.dex.spec.DexFile
+import com.linkedin.dex.spec.MethodAnnotation
+import com.linkedin.dex.spec.MethodIdItem
+
+/**
+ * Check if there are any class, field, method, or parameter annotations in the given class
+ *
+ * @return true if this [ClassDefItem] has any annotations, false otherwise
+ */
+fun hasAnnotations(classDefItem: ClassDefItem): Boolean {
+    return classDefItem.annotationsOff != 0
+}
+
+/**
+ * Null-safe creation of an [AnnotationsDirectoryItem]
+ */
+fun DexFile.getAnnotationsDirectory(classDefItem: ClassDefItem): AnnotationsDirectoryItem? {
+    if (hasAnnotations(classDefItem)) {
+        return AnnotationsDirectoryItem.create(byteBuffer, classDefItem.annotationsOff)
+    } else {
+        return null
+    }
+}
+
+/**
+ * @return A list of descriptors containing all class-level annotations in the given [AnnotationsDirectoryItem]
```

Comment:
Can we be consistent with referring to items either by name or by type? I see both here